content/doc/developer/security/csp.adoc

```diff
@@ -12,13 +12,68 @@ From https://content-security-policy.com/[content-security-policy.com]:
 Using Content-Security-Policy (CSP), injection attacks like cross-site scripting can be prevented.
 
 // TODO Keep up to date with versions
-Unfortunately, as of Jenkins 2.360, the Jenkins classic UI is not yet compatible with the CSP directives that would allow preventing such injection attacks.
+Unfortunately, as of Jenkins 2.372, the Jenkins classic UI is not yet compatible with the CSP directives that would allow preventing such injection attacks.
 
-This guide documents how to write UI code in a manner that is expected to be compatible with future use of CSP directives on the Jenkins UI.
+This guide documents how to identify components that will be incompatible with CSP rules and how to write and adapt UI code in a manner that is expected to be compatible with future use of CSP directives on the Jenkins UI.
 
+NOTE: Most of this page focuses on problems caused by the `script-src` CSP directive as that is the most common issue across Jenkins plugins and non-compliant code is most likely to result in link:/blog/2020/07/15/xss-severity/[severe vulnerabilities].
 
 
-== Guidelines
+== Statically Identifying Affected Code and Plugins
+
+To check plugins for CSP compatibility, look for the following code patterns:
+
+Inline `<script>` blocks::
+Referencing a file to load JavaScript from is fine, but inline block contents are not.
+All `<script>` tags without `src` are a problem.
+They will break without `'unsafe-inline'` in the `script-src` CSP directive.
```

Comment:
💡 I am not sure the "unsafe-inline" name is sufficiently explicit about its danger. Adding emphasis could be sort of a repetition :(

Reply:
This is developer docs, not admin docs.